mobile/lib/shared/providers/asset.provider.dart

```diff
@@ -9,24 +10,50 @@ import 'package:photo_manager/photo_manager.dart';
 
 class AssetNotifier extends StateNotifier<List<AssetResponseDto>> {
   final AssetService _assetService;
+  final AssetCacheService _assetCacheService;
+
   final DeviceInfoService _deviceInfoService = DeviceInfoService();
 
-  AssetNotifier(this._assetService) : super([]);
+  AssetNotifier(this._assetService, this._assetCacheService) : super([]);
+
+  _cacheState() {
+    _assetCacheService.put(state);
+  }
 
   getAllAsset() async {
+    final stopwatch = Stopwatch();
+
+
+    if (await _assetCacheService.isValid() && state.isEmpty) {
+      stopwatch.start();
+      state = await _assetCacheService.get();
+      debugPrint("Reading assets from cache: ${stopwatch.elapsedMilliseconds}ms");
+      stopwatch.reset();
+    }
```

Comment:
From my understanding of this code block, the data will be displayed from the cache first, then will fetch the new data in the background and update the state, correct?

Reply:
Yeah, the state is updated twice. First time from the cache, second time from API.
I think one possible optimization would be to start both background jobs first and then await the results. But that would cause a race condition so I avoided it.

Reply:
Got it, would it worth to create an api endpoint to get the asset count for each user and check that count before perform the fetch from the server?

Reply:
IMO not really. Checking the cache costs as few ms, loading data from the cache 200-300ms.